File: 219.py
class Solution:
    def containsNearbyDuplicate(self, nums: list, k: int) -> bool:
        map = {}
        for i, num in enumerate(nums):
            if num in map and i - map[num] <= k:
                return True
            else: map[num] = i
        return False



if __name__ == "__main__":
    s = Solution()
    print(s.containsNearbyDuplicate([1,2,3,1], 3))

File: 219.go
package leetcode

func containsNearbyDuplicate(nums []int, k int) bool {
	mp := map[int]int{}
	n := len(nums)
	for i := 0; i < n; i++ {
		if _, ok := mp[nums[i]]; ok {
			return true
		}
		mp[nums[i]] = i
		if j := i - k; j >= 0 {
			if mp[nums[j]] == j {
				delete(mp, nums[j])
			}
		}
	}
	return false
}
